crates/driver/src/infra/api/error.rs

```diff
@@ -0,0 +1,104 @@
+use {
+    crate::{
+        domain::competition::{self, quote, solution},
+        infra::api,
+    },
+    hyper::StatusCode,
+    serde::Serialize,
+};
+
+#[derive(Debug, Clone, Copy, Serialize)]
+#[serde(rename_all = "PascalCase")]
+enum Type {
+    QuotingFailed,
+    SolverFailed,
+    SolutionNotFound,
+    DeadlineExceeded,
+    SimulationFailed,
+    Unknown,
+    TransactionPublishingFailed,
+    InvalidAuctionId,
+    MissingSurplusFee,
+    LiquidityError,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Error {
+    error_type: Type,
```

Comment:
nit: `error.error_type` feels redundant, I would suggest:
```suggestion
    kind: Type,
```

Reply:
https://github.com/cowprotocol/services/pull/1128/commits/abda069a433854826d5d895ec5ddfbc901532527